firestore/query.go

```diff
@@ -569,8 +579,48 @@ func trunc32(i int) int32 {
 	return int32(i)
 }
 
+// Get returns an array of query's resulting documents.
+func (q Query) Get(ctx context.Context) ([]*DocumentSnapshot, error) {
```

Comment:
To be consistent with Query.Documents , I wonder if this should be called `GetAll`

Reply:
Looks like pretty much all other get methods in cloud.google.com/go/firestore which return a slice are called GetAll, whereas Get() is used for a single Transaction or DocumentRef. From that perspective I think GetAll makes more sense.

Also, I feel like the fact that this method is being added to the surface for the library is a bigger change than what's implied by the title of the PR. Could you change the title to reflect this? I also think it would be good to add an example of how this would be used to examples_test.go for the docs.

Reply:
One more thing-- looking at this a little more closely I see that the following are now equivalent for a non-limitToLast query `q`:

`q.Get()`
`q.Documents().GetAll()`

Do we have an opinion about which is preferred?

And really, is it necessary to have this duplication? (I guess the answer is yes with the current design because DocumentIterator doesn't have access to the Query fields. But in theory we could just add an unexported field there and pass it along I think-- this would mean that to do a limitToLast query you would also call `q.Documents().GetAll()`, though it would mean that other DocumentIterator methods would then need to error if limitToLast were set to true).

Would be curious on your thoughts @benwhitehead @codyoss

Reply:
If there isn't a semantic difference to how those methods execute, I suspect it's fine to have a single method. And if there is a uniform way for users to keep doing what they're used to doing to read results it's probably better to stick with that, than it is to try and have this implementation be uniform with the other languages. 

The complexity with `limitToLast` is that it has to buffer the up to `n` results and flip the order in memory before return to the application. So, as long as that property can be ensured whichever way is fine with me.

Reply:
Discussed this with @codyoss offline, he also thinks that removing Query.Get and enabling this feature through the existing Query.DocumentIterator.GetAll method instead seems preferable. @IlyaFaer could you modify this change to do that? Feel free to reach out if there are questions that come up.

Reply:
NP. As you mentioned before, I've added an unexported `Query` field into `DocumentIterator` and moved all the implementation into `DocumentIterator.GetAll()`. Seems to me we're no more in need of an example (as there are no new methods added), and a separate test.